client/my-sites/sharing/connections/service.jsx

```diff
@@ -80,49 +70,32 @@ const SharingService = React.createClass( {
 				this.props.connections.once( 'connect', function() {
 					if ( serviceConnections.didKeyringConnectionSucceed( service.ID, this.props.siteId ) &&
 						serviceConnections.isServiceForPublicize( service.ID ) ) {
-						this.setState( { selectingAccountForService: service } );
+						this.setState( { isSelectingAccount: true } );
 					}
 				}.bind( this ) );
 			} else {
-				analytics.ga.recordEvent( 'Sharing', 'Clicked Connect Button in Modal', this.state.selectingAccountForService.ID );
+				this.props.recordGoogleEvent( 'Sharing', 'Clicked Connect Button in Modal', this.props.service.ID );
 			}
 		} else {
 			// If an account wasn't selected from the dialog or the user cancels
 			// the connection, the dialog should simply close
 			this.props.connections.emit( 'create:error', { cancel: true } );
-			analytics.ga.recordEvent( 'Sharing', 'Clicked Cancel Button in Modal', this.state.selectingAccountForService.ID );
+			this.props.recordGoogleEvent( 'Sharing', 'Clicked Cancel Button in Modal', this.props.service.ID );
 		}
 
 		// Reset active account selection
-		this.setState( { selectingAccountForService: null } );
+		this.setState( { isSelectingAccount: null } );
 	},
 
 	toggleSitewideConnection: function( connection, isSitewide ) {
 		this.props.connections.update( connection, { shared: isSitewide } );
 	},
 
-	getAccountDialog: function() {
-		const isSelectingAccount = !! this.state.selectingAccountForService;
-		let accounts;
-
-		if ( isSelectingAccount ) {
-			accounts = serviceConnections.getAvailableExternalAccounts( this.state.selectingAccountForService.ID, this.props.siteId );
-		}
-
-		return (
-			<AccountDialog
-				isVisible={ isSelectingAccount }
-				service={ this.state.selectingAccountForService }
-				accounts={ accounts }
-				onAccountSelected={ this.addConnection } />
-		);
-	},
-
 	onConnectionSuccess: function() {
 		this.setState( { isConnecting: false } );
 		this.props.connections.off( 'create:error', this.onConnectionError );
 
-		notices.success( this.translate( 'The %(service)s account was successfully connected.', {
+		notices.success( this.props.translate( 'The %(service)s account was successfully connected.', {
```

Comment:
Only `onConnectionError` uses it more than once and those callback strings will go into the notice middleware eventually. Would it be okay to keep it for now?